### DISCUSSION TITLE
Update Needed message and locked out of Nextcloud

### DISCUSSION THREAD

**Discussion by cyberjock** [nextcloud/all-in-one#1459] [Questions] [answered]
Hello all.  I'm new to using docker (and linux) but I have extensive experience with FreeBSD and jails.  About a month or so ago I created my nextcloud AIO VM on Proxmox and it's been running smoothly.  Sometime since last Wednesday my Nextcloud AIO environment seems to have changed.  Now when I visit my nextcloud page I get a "Update needed.  Please use the command line updater because automatic updating is disabled in the config.php".

I've never seen this before, and Googling finds other people that have had this issue years ago, but no real explanation for why it suddenly happened for them (or me).  I did do an update about 10 or so days ago, so I'm a bit surprised that I'd be "so out of date that the software is suddenly broken".

AIO webpage shows I'm on 3.0.0, but when I look at stuff like this https://github.com/nextcloud/all-in-one/commit/9e0079effc1e561cd9adacefc5af5e94617dae17 I'm left wondering if I'm actually behind.  I'm really not sure where to go from here, but I'd like to get my Nextcloud environment working properly.

If I understand things properly, I'm not actually supposed to update nextcloud per the command line with occ commands inside of the docker container because updates really should come from docker images themselves.  Of course, if I'm wrong feel free to call me out on it!

At this stage, I'm looking for 2 things:
1.  How do I get out of this situation so I can start using my nextcloud environment again.  (and if anyone knows why this happened to begin with, I'd love to understand why)
2.  I regularly update the VM and reboot it via apt.  Is there a CLI command to tell docker to shutdown all of the containers and update them?  I figure it's probably easier to simply do this from the CLI once a week when I do apt updates than wait until my nextcloud files are no longer accessible.  Since the VM itself is running ZFS, I was thinking about also doing ZFS snapshots as a form of backup when doing upgrades from the CLI.

Thanks!

**Comment by szaimen** [comment]
Can you post the update logs here?
`sudo docker exec -it nextcloud-aio-nextcloud cat /var/www/html/data/update.log`

**Reply by cyberjock** [reply]
root@nextcloud:~# sudo docker exec -it nextcloud-aio-nextcloud cat /var/www/html/data/update.log
cat: can't open '/var/www/html/data/update.log': No such file or directory

I tried a few variations just to be safe, but still nothing.

```
root@nextcloud:~# sudo docker exec -it nextcloud-aio-nextcloud "cat /var/www/html/data/update.log"
OCI runtime exec failed: exec failed: unable to start container process: exec: "cat /var/www/html/data/update.log": stat cat /var/www/html/data/update.log: no such file or directory: unknown
root@nextcloud:~# sudo docker exec -it nextcloud-aio-nextcloud 'cat /var/www/html/data/update.log'
OCI runtime exec failed: exec failed: unable to start container process: exec: "cat /var/www/html/data/update.log": stat cat /var/www/html/data/update.log: no such file or directory: unknown
```

I know what you're trying to do, but I'm not a docker expert (this is my first docker container) so I'm doing this "trial by fire". So I'm not sure how to get inside the container to get that file for you.

**Reply by szaimen** [reply]
Ah right, sorry my fault.

**Reply by cyberjock** [reply]
Okay, I tried this:

```
root@nextcloud:~# docker exec -it nextcloud-aio-nextcloud /bin/bash
bash-5.1# find / -name "update.log"
bash-5.1#
```

So no update.log file seems to exist in the container.

**Reply by szaimen** [reply]
Yes, as I said my fault.

**Comment by szaimen** [comment, marked as answer]
You can check if that works:
`sudo docker exec --user www-data -it  nextcloud-aio-nextcloud php occ upgrade`

**Reply by cyberjock** [reply]
```
root@nextcloud:~# sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Setting log level to debug
Turned on maintenance mode
Updating database schema
Updated database
Updating <files_antivirus> ...
Updated <files_antivirus> to 4.0.0
Update app deck from App Store
Starting code integrity check...
Finished code integrity check
Update successful
Turned off maintenance mode
Resetting log level
```

Okay, so it's now back up and working.  So please answer a few questions for me:

1.  Isn't this "not the right way" to update nextcloud in an AIO situation?
2.  If so, is this going to go badly on the next "official update"?
3. Is there a recommended way to update all of the docker containers from the CLI?  Obviously this doesn't seem to be something that would have been prevented in this particular situation.  I just like to streamline my processes from the CLI when I decide to do updates.

Thanks.

**Reply by szaimen** [reply]
I think that one of your installed Nextcloud apps triggered this. We will improve this with 3.1.0 which will get released soon so that it hopefully does not happen anymore in the future. V3.1.0 will also populate the updater log so that we can check what went wrong during an update.

**Reply by RomanSonnik** [reply]
Hello, how much do I need to wait when "Starting code integrity check..."?

**Reply by RomanSonnik** [reply]
oups nevermind.

**Comment by cyberjock** [comment]
Quick update for anyone that looks at this in the future:

1. I'm getting this error in Nextcloud now:
```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => 94195a260a005dac543c3f6aa504f1b28e0078297fe94a4f52f012c16c109f0323eecc9f767d6949f860dfe454625fcaf1dc56f87bb8350975d8f006bbbdf14a
                            [current] => 1b07fb272efa65a10011ed52a6e51260343c5de2a256e1ae49f180173e2b6684ccf90d1af3c19fa97c31d42914866db46e3216883ec0d6a82cec0ad5529e78b1
                        )

                )

        )

)
```

I presume this is "expected" until the next release of Nextcloud AIO comes out considering my problems.

I investigated this file, and it looks like its tied to Open Office addon.  Apparently I'm not the only one to have this issue, although I seem to be the only one this month... https://github.com/ONLYOFFICE/onlyoffice-nextcloud/issues/600

As the machine running Nextcloud is running ZFS, I did a ZFS scrub to check for file system corruption and none was found.  So I'm pretty sure it's not due to any file system issues.

Thanks!

**Reply by szaimen** [reply]
Yes, this should be fixed with the next aio release comes out 👍
